### PR TITLE
SBP daemon improvements (master)

### PR DESCRIPTION
**Pull request by jacobmcnamee**
- Add init script to extract SBP sender ID from `/factory/uuid` and write it to `/cfg/sbp_sender_id`
- Update `piksi_system_daemon`, `sbp_settings_daemon`, and `sbp_log` to read SBP sender ID from `/cfg/sbp_sender_id`
- Move fileio to `sbp_fileio_daemon` and launch two instances - one for internal use and one for external use (prevents internal fileio messages from being sent out externally).
- Update routing rules and monitrc

Resolves https://github.com/swift-nav/piksi_v3_bug_tracking/issues/175

TODO:
- [x] Rebase once #83 is merged
- [x] Update firmware https://github.com/swift-nav/piksi_firmware_private/pull/446
- [x] Test reset
- [x] Test file IO
- [ ] Test sender ID

/cc @swift-nav/firmware

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v0.3.0-31-gf81f3c8
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v0.3.0-31-gf81f3c8/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v0.3.0-31-gf81f3c8/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v0.3.0-31-gf81f3c8

**Comment by jacobmcnamee** [comment]
The settings daemon needs to send messages with sender ID = `SBP_SENDER_ID` or they are ignored by users.

File: board/piksiv3/rootfs-overlay/etc/init.d/S18cfg [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+#
+# initialize config data
+#
+
+case "$1" in
+  start)
+  mkdir -p /cfg
+  SBP_SENDER_ID_HEX=`cat /factory/uuid | tail -c 4`
+  printf "%u" "0x$SBP_SENDER_ID_HEX" > /cfg/sbp_sender_id
```
Comment (fnoble) [inline board/piksiv3/rootfs-overlay/etc/init.d/S18cfg]: What about putting this in `/var/cfg/sbp_sender_id`?

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-9-g43a241e
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-9-g43a241e/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-9-g43a241e/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-9-g43a241e

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-9-g0037f16
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-9-g0037f16/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-9-g0037f16/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-9-g0037f16

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-14-g7ab514b
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-14-g7ab514b/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-14-g7ab514b/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-14-g7ab514b

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-14-g7ab514b
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-14-g7ab514b/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-14-g7ab514b/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-14-g7ab514b

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-14-ga8990dd
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-14-ga8990dd/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-14-ga8990dd/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-14-ga8990dd

File: board/piksiv3/rootfs-overlay/etc/init.d/S18cfg (hunk 1 above)
Comment (jacobmcnamee) [inline board/piksiv3/rootfs-overlay/etc/init.d/S18cfg]: >Variable files—files whose content is expected to continually change during normal operation of the system—such as logs, spool files, and temporary e-mail files.

I'm not convinced `/var` is the place. Maybe we should consolidate `/factory`, `/img_tbl`, and `/cfg` under `/piksi` or similar.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-20-g7553e5f
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-20-g7553e5f/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-20-g7553e5f/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-20-g7553e5f

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-20-gcf27574
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-20-gcf27574/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-20-gcf27574/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-20-gcf27574

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-24-g590492f
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-24-g590492f/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-24-g590492f/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-24-g590492f

**Pull request merged**